File: probs/usaco/oldsite/folding/folding.cpp
// solution by Albert Ye
// AlphaStar 2020 Summer Camp: CC31AB
// Problem: feb12-folding
#include <bits/stdc++.h>

using namespace std;

// number of knots, length (not needed), position, difference between adjacent positions
int n, l, pos[105], diff[105];
int main()
{
	cin >> n >> l;
	for(int i = 0; i < n; i++) cin >> pos[i];
	sort(pos, pos+n); // sort the positions
	// and find the differences accordingly
	for(int i = 0; i < n-1; i++)
	{
		diff[i] = pos[i+1] - pos[i];
	}
	int ans = 0; // number of palindromes
	// check for palindromes on left and right
	for(int i = 0; i < n-1; i++)
	{
		bool works = 1;
		// 0 to i
		for(int j = 0; j <= i - j; j++)
		{
			if(diff[j] != diff[i-j])
			{
				works = 0;
				break;
			}
		}
		ans+=works;
	}
	for(int i = 1; i < n-1; i++)
	{
		bool works = 1;
		// i to n-2
		for(int j = 0; i + j <= n-2 - j; j++)
		{
			if(diff[i+j] != diff[n-2 - j])
			{
				works = 0;
				break;
			}
		}
		ans+=works;
	}
	cout << ans << endl;
}

File: probs/usaco/oldsite/folding/folding.py
"""
Solution by Albert Ye
AlphaStar 2020 Summer Camp: CC31AB
Problem: feb12-folding
"""
n, l = input().split(' ')
n = int(n) # number of knots in rope
l = int(l) # length of rope
pos = [] # position of knots in rope
diff = [] # distance between adjacent knots
for i in range(n) :
    pos.append(int(input()))
pos.sort()
for i in range(n-1) :
    diff.append(pos[i+1] - pos[i])
"""
Key observation is that valid positions correspond
to palindromes in the diff array.
"""
# number of valid positions
ans = 0
# check for palindromes on the left or the right
# left palindromes
for i in range(n-1) :
    works = True
    for j in range(n) :
        if j > i - j :
            break
        if diff[j] != diff[i-j] :
            works = False
            break
    if works :
        ans += 1
# right palindromes
for i in range(1, n-1) :
    works = True
    for j in range(n) :
        if i + j > n-2 - j :
            break
        if diff[i+j] != diff[n-2-j] :
            works = False
            break
    if works :
        ans += 1
print(ans)
